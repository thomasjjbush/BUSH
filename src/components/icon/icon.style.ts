import styled from 'styled-components';
import { IconProps as Props } from '../../types';
import { StyledButton } from '../button/button.style';

const shared = (color: string, icon: string, size: number): string => `
	display: flex;
	font-family: 'icomoon' !important;
	speak: never;
	font-style: normal;
	font-weight: normal;
	font-variant: normal;
	text-transform: none;
	line-height: 1;
	-webkit-font-smoothing: antialiased;
	-moz-osx-font-smoothing: grayscale;
	color: ${color};
	font-size: ${size || 20}px;

	:after {
		content: "\\${icon}";
		flex: 1;
		display: flex;
		justify-content: center;
		align-items: center;
  	}
`;

export const ClickableIcon = styled(StyledButton)<Props>`
    ${(props): string => shared(props.color || props.theme.colors.brand, props.icon, props.size)}
    align-self: auto;
    appearance: none;
    background: transparent;
    border: 0;
    margin: 0;
    padding: 0;

    &[disabled] {
        cursor: not-allowed;
        opacity: 0.4;
    }
`;

export const Icon = styled.i<Props>`
    ${(props): string => shared(props.color || props.theme.colors.brand, props.icon, props.size)}
`;
