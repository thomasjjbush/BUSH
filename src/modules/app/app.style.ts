import styled from 'styled-components';
import { StyledButton } from '../../components/button/button.style';

export const StyledToggle = styled.div`
    bottom: 30px;
    right: 30px;
    position: absolute;
    z-index: 5;
`;
