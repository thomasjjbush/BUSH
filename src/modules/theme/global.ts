import styled, { createGlobalStyle } from 'styled-components';
import { StyledProps } from '../../types';

export const Global = createGlobalStyle<StyledProps>`
    html, body {
        height: 100%;
        margin: 0;
        overflow: hidden;

        #root {
            background-color: ${(props): string => props.theme.colors.background};
            display: flex;
            flex-direction: column;
            height: inherit;
            position: relative;
        }
    }

    * {
        box-sizing: border-box;
        color: ${(props): string => props.theme.colors.text};
        text-decoration: none;
        
        @media not all and (hover: none) {
            &:focus {
                outline: dashed 2px ${(props): string => props.theme.colors.brand};
                outline-offset: 2px;
            }
        } 

        @media (hover: none) {
            &:focus {
                outline: none;
            }
        }
    }

    section {
        flex: 1;
        overflow: inherit;

        &:not(:last-of-type) {
            margin-right: 30px;
        }
    }

    h1, h2, h3, h4, h5, h6 {
        font-family: 'Anton', sans-serif;
        margin: 0;
    }

    h1 {
        font-size: 45px;

        @media only screen and (max-width: ${(props): number => props.theme.breakpoints.mobile}px) {
            font-size: 10vw;
        }
    }

    a, button, input, p, select, table {
        font-family: 'Raleway', sans-serif;
    }

    p {
        
        font-size: 20px;
    }

    @font-face {
        font-family: 'icomoon';
        src: url(https://assets.ctfassets.net/e85zpqq4b2pc/kUqasbyE…NaEeXK/e45c4da93623e96f9c35d2923dd226c7/icons.eot);
        src: url('https://assets.ctfassets.net/e85zpqq4b2pc/kUqasbyE…NaEeXK/e45c4da93623e96f9c35d2923dd226c7/icons.eot?#iefix') format('embedded-opentype'), 
             url(https://assets.ctfassets.net/e85zpqq4b2pc/3QQ5NdjvE2YSOArgeYMEhx/11eae45950e74136401fc1cc2b4509a7/icons.woff) format('woff'), 
             url(https://assets.ctfassets.net/e85zpqq4b2pc/4dcDVFnRJrQrZ8BDiHTd8F/e5ff802efc33a2cc4e1d1879c622327d/icons.ttf) format('truetype'),
             url('https://images.ctfassets.net/e85zpqq4b2pc/5VBXsOEP…lwUccQ/f65bad72e6323950c1959e498ade9e97/icons.svg#icons') format('svg');
      }
`;

export const StyledMain = styled.main<StyledProps<{ max?: number; padding?: number; vertical?: boolean }>>`
    display: flex;
    flex-direction: ${(props): string => (props.vertical ? 'column' : 'auto')};
    flex: 1;
    overflow: hidden;
    padding: ${(props): number => props.padding || 0}px;
    margin: ${(props): string => (props.max ? 'auto' : 'none')};
    max-width: ${(props): string => (props.max ? `${props.max}px` : 'none')};
    width: 100%;

    @media only screen and (max-width: ${(props): number => props.theme.breakpoints.tabletPortrait}px) {
        flex-direction: column;
    }
`;
