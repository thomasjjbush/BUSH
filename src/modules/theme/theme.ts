import { Theme } from '../../types';
import { isWebpSupported } from '../../utils';

const pallette = {
    black: '#080808',
    darkGrey: '#1a1a1a',
    lightGrey: '#eaeaea',
    pink: '#da335c',
    white: '#ffffff',
};

export const initTheme = (darkMode: boolean): Theme => ({
    breakpoints: {
        mobile: 600,
        tabletLandscape: 1025,
        tabletPortrait: 769,
    },
    colors: {
        highlight: darkMode ? pallette.black : pallette.white,
        background: darkMode ? pallette.darkGrey : pallette.white,
        border: pallette.lightGrey,
        brand: pallette.pink,
        offset: darkMode ? pallette.black : pallette.lightGrey,
        text: darkMode ? pallette.white : pallette.black,
    },
    images: {
        logo: {
            pink:
                'https://images.ctfassets.net/e85zpqq4b2pc/6fw1X0NtyqbS5p26FT7sJc/9b16e3c639c6030c3f40e90aea5fb5e6/logo-pink-v2.png',
            white:
                'https://images.ctfassets.net/e85zpqq4b2pc/7FIScBWfqqMSdu5Cq8J99B/373ef358e312947a14bce330fe933ae9/logo-white-v2.png',
        },
        loading:
            'https://images.ctfassets.net/e85zpqq4b2pc/7oqaXxGJ4OYVMunVaJR0qA/bba0a34f6717ce60b3361dd86b0d633f/loading.gif',
    },
    // isWebpSupported: isWebpSupported(),
});
