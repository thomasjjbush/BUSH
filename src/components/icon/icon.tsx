import React, { FunctionComponent, ReactElement } from 'react';
import { IconProps as Props } from '../../types';
import { StyledIcon } from './icon.style';

export const Icon: FunctionComponent<Props> = (props: Props): ReactElement => (
    <StyledIcon data-test-id={props.testID || 'icon'} {...props} {...(props.onClick && { tabIndex: 0 })} />
);

Icon.displayName = 'Icon';
