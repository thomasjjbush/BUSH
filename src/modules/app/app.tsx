import React, { FunctionComponent, ReactElement, Suspense, useMemo, useState } from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import { Icon, Loading, Nav } from '../../components';
import { Global } from '../theme/global';
import { initTheme } from '../theme/theme';
import { routes } from './../../config/';
import { StyledToggle } from './app.style';
import { Icons } from '../../types';
import { client, GraphQLContext } from '../../utils';

export const App: FunctionComponent = (): ReactElement => {
    const [darkMode, setDarkMode] = useState(true);
    const theme = useMemo(() => initTheme(darkMode), [darkMode]);

    return (
        <ThemeProvider theme={theme}>
            <GraphQLContext.Provider value={client}>
                <Global />
                <Router>
                    <Nav darkMode={darkMode} />
                    <Suspense fallback={<Loading />}>
                        <Switch>
                            {routes.map(({ path, module: Module }) => (
                                <Route exact key={path} path={path}>
                                    <Module />
                                </Route>
                            ))}
                        </Switch>
                    </Suspense>
                    <aside>
                        <StyledToggle>
                            <Icon
                                ariaLabel={`Enable ${darkMode ? 'light' : 'dark'} mode`}
                                color={theme.colors.text}
                                icon={Icons.CONTRAST}
                                onClick={(): void => setDarkMode(!darkMode)}
                                size={50}
                                testID="contrast-button"
                            />
                        </StyledToggle>
                    </aside>
                </Router>
            </GraphQLContext.Provider>
        </ThemeProvider>
    );
};
