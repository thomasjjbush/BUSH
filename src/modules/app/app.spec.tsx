import React from 'react';
import { shallow } from 'enzyme';
import { App } from './app';
import { initTheme } from '../theme/theme';
import { Icon } from '../../components/icon/icon';

jest.mock('../theme/theme', () => ({ initTheme: jest.fn().mockReturnValue({ colors: { text: 'colors.text' } }) }));
jest.mock('../../utils', () => ({
    ...jest.requireActual('../../utils'),
    client: 'client',
}));

describe('App', () => {
    beforeEach(() => {
        jest.clearAllMocks();
    });

    it('should match snapshot', () => {
        expect(shallow(<App />)).toMatchSnapshot();
    });

    it('should toggle dark mode on toggle click', () => {
        const wrapper = shallow(<App />);
        expect(initTheme).toHaveBeenCalledTimes(1);
        expect(initTheme).toHaveBeenCalledWith(true);
        expect(wrapper.find(Icon).prop('ariaLabel')).toBe('Enable light mode');

        wrapper.find(Icon).prop('onClick')();
        expect(initTheme).toHaveBeenCalledTimes(2);
        expect(initTheme).toHaveBeenNthCalledWith(2, false);
        expect(wrapper.find(Icon).prop('ariaLabel')).toBe('Enable dark mode');
    });
});
