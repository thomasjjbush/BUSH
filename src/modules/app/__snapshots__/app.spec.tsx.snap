// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<Component
  theme={
    Object {
      "colors": Object {
        "text": "colors.text",
      },
    }
  }
>
  <ContextProvider
    value="client"
  >
    <Memo(l) />
    <BrowserRouter>
      <Nav />
      <Suspense
        fallback={<Unknown />}
      >
        <Switch>
          <Route
            exact={true}
            key="/"
            path="/"
          >
            <lazy />
          </Route>
          <Route
            exact={true}
            key="/projects/:slug"
            path="/projects/:slug"
          >
            <lazy />
          </Route>
        </Switch>
      </Suspense>
      <aside>
        <styled.div
          styles="
    bottom: 30px;
    right: 30px;
    position: absolute;
    z-index: 5;
"
        >
          <Icon
            ariaLabel="Enable light mode"
            color="colors.text"
            icon="e9d5"
            onClick={[Function]}
            size={50}
            testID="contrast-button"
          />
        </styled.div>
      </aside>
    </BrowserRouter>
  </ContextProvider>
</Component>
`;
